src/app/search/search.component.html

```diff
@@ -1,25 +1,104 @@
-      <div *ngFor="let product of products" (click)='toProductDetail(product.id)' data-id="product.id">
-        <div class="card m-3">
-          <img [src]="product.icon" class="card-img-top" alt="icon">
-          <div class="card-body">
-            <h1 class="card-title">{{product.title}}</h1>
-            <p class="card-text">{{product.teaser}}</p>
-          </div>
+
+    <hr>
+
+    <div class="row d-flex justify-content-center mt-5">
+      <div class="col d-flex justify-content-between align-items-end">
+
+        <div *ngIf="selectedCategory" class="d-flex">
+          <h2>{{this.filteredProducts.length}} Results for "{{selectedCategory}}" category</h2>
+        </div>
+
+        <div *ngIf="selectedStatus" class="d-flex">
+          <h2>{{this.filteredProducts.length}} Results for "{{selectedStatus}}" status</h2>
         </div>
+
+        <div *ngIf="!selectedCategory && !selectedStatus" class="d-flex">
+          <h2>{{this.searchCount}} Results for "{{this.query}}"</h2>
+        </div>
+
+        <div class="d-flex">
+          <mat-form-field appearance="fill">
+            <mat-label>Release date: </mat-label>
+            <mat-select [formControl]="releaseDate" [(value)]="selectedReleaseDate">
+              <mat-select-trigger>
+                {{releaseDate.value ? selectedReleaseDate : ''}}
+              </mat-select-trigger>
+              <mat-option *ngFor="let releaseDate of releaseDateList" [value]="releaseDate">{{releaseDate}}</mat-option>
+            </mat-select>
+          </mat-form-field>
+        </div>
+
+      </div>
+    </div>
+
+    <div class="top-100 row d-flex justify-content-center">
+      <div class="col d-flex justify-content-between align-items-start"
+            *ngFor="let product of visibleProducts"
+            (click)='toProductDetailPage(product.id)'
+            data-id="product.id">
+
+        <mat-card class="mat-card card">
+          <mat-card-header>
+            <mat-card-title>{{product.title}}</mat-card-title>
+          </mat-card-header>
+          <img mat-card-image [src]="product.image" alt="Photo of a Shiba Inu">
+          <mat-card-content>
+            <p>{{product.teaser}}</p>
+          </mat-card-content>
+          <mat-card-actions>
+          </mat-card-actions>
+        </mat-card>
+
       </div>
     </div>
+
+    <div *ngIf="selectedCategory || selectedStatus">
+      <mat-paginator [length]="this.filteredProducts.length"
+                      [pageSize]="12"
+                      aria-label="Select page"
+                      (page)="onPageChange($event, this.filteredProducts)">
+      </mat-paginator>
+    </div>
+
+    <div *ngIf="!selectedCategory && !selectedStatus">
+      <mat-paginator [length]="this.allProductsBySearch.length"
+                      [pageSize]="12"
+                      aria-label="Select page"
+                      (page)="onPageChange($event, this.allProductsBySearch)">
+      </mat-paginator>
+    </div>
   </div>
 </div>
```

Comment:
Still there is lots of bootstrap in this file

Reply:
I replaced bootstrap with Angular Material (Grid-list for layout)